core/key.go

```diff
@@ -291,22 +292,28 @@ func NewCollectionSchemaVersionKeyFromString(key string) (CollectionSchemaVersio
 }
 
 // NewCollectionIndexKey creates a new CollectionIndexKey from a collection name and index name.
-func NewCollectionIndexKey(colID, indexName string) CollectionIndexKey {
-	return CollectionIndexKey{CollectionName: colID, IndexName: indexName}
+func NewCollectionIndexKey(colID immutable.Option[uint32], indexName string) CollectionIndexKey {
```

Comment:
question: Related to my question above, since ID is optional (due to I guess name being optional) then should probably change on `Collection interface` as well?

Reply:
I think you misunderstand, if `colID` here is optional it will create an empty key that allows scanning over the entire prefix.  `CollectionDescription.ID` is not optional.